### PR TITLE
feat: Added cluster scope manifests. Fixes #498

### LINKED ISSUE #498: Cluster-scoped manifests
We need to provide a way to install Dataflow at cluster scope.

### PR DESCRIPTION
**Pull request by sarabala1979**

### CODE REVIEW

File: config/cluster-rbac/leader_election_role.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# permissions to do leader election.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: leader-election-role
```
Comment (alexec) [inline config/cluster-rbac/leader_election_role.yaml:5]: not needed?

File: config/cluster-rbac/pipeline-role.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
```
Comment (alexec) [inline config/cluster-rbac/pipeline-role.yaml:2]: duplicate?

File: config/cluster-rbac/pipeline-rolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
```
Comment (alexec) [inline config/cluster-rbac/pipeline-rolebinding.yaml:2]: duplicate?

File: config/cluster-rbac/pipeline-sa.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: ServiceAccount
```
Comment (alexec) [inline config/cluster-rbac/pipeline-sa.yaml:2]: duplicate?

**Review by alexec — CHANGES_REQUESTED** [review]
@sarabala1979 I think there might be some duplication here as I'd expect only to see new files that are `Cluster*` but I see things like `Role`. To avoid having to invest too much time maintaining this, can I ask you to double-check for duplicates?

File: config/cluster-install/controller-manager-deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+        - name: manager
+          env:
+            - name: ARGO_DATAFLOW_CLUSTER
+              value: quick-start
```
Comment (alexec) [inline config/cluster-install/controller-manager-deploy.yaml:13]: this is wrong, user must specify

File: config/cluster-rbac/auth_proxy_client_clusterrole.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (alexec) [inline config/cluster-rbac/auth_proxy_client_clusterrole.yaml:1]: maybe put this into `cluster-install/rbac`?

File: config/cluster-rbac/auth_proxy_service.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Service
```
Comment (alexec) [inline config/cluster-rbac/auth_proxy_service.yaml:2]: dupe?

File: config/cluster-rbac/leader_election_role_binding.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (alexec) [inline config/cluster-rbac/leader_election_role_binding.yaml:1]: dupe?

**Comment by sarabala1979** [comment]
> @sarabala1979 I think there might be some duplication here as I'd expect only to see new files that are `Cluster*` but I see things like `Role`. To avoid having to invest too much time maintaining this, can I ask you to double-check for duplicates?

Sorry, I didn't locally by I didn;t do git rm<>. I will fix it

File: config/cluster-rbac/auth_proxy_client_clusterrole.yaml (hunk 6 above)
Comment (sarabala1979) [reply config/cluster-rbac/auth_proxy_client_clusterrole.yaml]: moved cluster-install into quick-cluster-install

File: config/cluster-install/controller-manager-deploy.yaml (hunk 5 above)
Comment (sarabala1979) [reply config/cluster-install/controller-manager-deploy.yaml]: So User needs to edit the YAML before applying it.

Comment (sarabala1979) [reply config/cluster-install/controller-manager-deploy.yaml]: Can we have it default for `quick-cluster-start.yaml` file?

File: config/default-cluster/pipeline-role.yaml [hunk 9]
```diff
@@ -0,0 +1,80 @@
+apiVersion: v1
```
Comment (alexec) [inline config/default-cluster/pipeline-role.yaml:1]: can we have :

* one resource per file
* file always named `{metadata.name}-{kind.ToLowerCase}.yaml`

**Review by alexec — CHANGES_REQUESTED** [review]

File: config/default/kustomization.yaml [hunk 10]
```diff
@@ -17,14 +17,16 @@ bases:
 #- ../prometheus
 
 resources:
-  - ssh-configmap.yaml
-  - lead-replica-priorityclass.yaml
+  - ../base-patch/ssh-configmap.yaml
```
[the remark this answers is not in the document]
Comment (sarabala1979) [reply config/default/kustomization.yaml:20]: These are common files for. Cluster and namespaced installation

File: config/default-cluster/pipeline-role.yaml [hunk 11]
```diff
@@ -0,0 +1,63 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: pipeline-role
```
Comment (alexec) [inline config/default-cluster/pipeline-role.yaml:4]: I don't think we should suffix names with the kind like this, can you remove `-role`?

File: config/default-cluster/rbac/cluster_role.yaml [hunk 12]
```diff
@@ -0,0 +1,70 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: manager-role
```
Comment (alexec) [inline config/default-cluster/rbac/cluster_role.yaml:4]: as above

File: config/default-cluster/rbac/cluster_role_binding.yaml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: manager-rolebinding
```
Comment (alexec) [inline config/default-cluster/rbac/cluster_role_binding.yaml:4]: as above

**Review by alexec — APPROVED** [review]

**Pull request merged**